### ISSUE TITLE
Reproduce Result in Kitti dataset

### ISSUE THREAD

**Issue by ftlong6666** [Nicholasli1995/EgoNet#6]
Hello @Nicholasli1995 thank you for your implementation. I have several questions.
1) What split (train and validation split) did you use for this code? is it `subcnn` split?
2) during the inference using testing set do you use ROI from the object detection algorithm?
3) Is it still use the calibration or this system can automatically generate it?
4) How to produce the similar table as in paper? I see that you use AOS, AP, also something like Easy, Medium, Hard. However when I tried to use validation set evaluation and use `kitti_eval_offline` I only got some plots, and statistic informations in txt and final result as here.

```
python inference.py --cfg "../configs/try_KITTI_inference:test_submission.yml"

Wrote prediction file at ../result/gt_box_test/data/007025.txt
Warning: 007025.png not included in detected images!
PDFCROP 1.38, 2012/11/02 - Copyright (c) 2002-2012 by Heiko Oberdiek.
==> 1 page written on `car_detection.pdf'.
PDFCROP 1.38, 2012/11/02 - Copyright (c) 2002-2012 by Heiko Oberdiek.
==> 1 page written on `car_orientation.pdf'.
PDFCROP 1.38, 2012/11/02 - Copyright (c) 2002-2012 by Heiko Oberdiek.
==> 1 page written on `car_detection_ground.pdf'.
PDFCROP 1.38, 2012/11/02 - Copyright (c) 2002-2012 by Heiko Oberdiek.
==> 1 page written on `car_detection_3d.pdf'.
Thank you for participating in our evaluation!
Loading detections...
number of files for evaluation: 1634
  done.
save ../result/submission/plot/car_detection.txt
car_detection AP: 93.934082 84.951851 67.594185
save ../result/submission/plot/car_orientation.txt
car_orientation AP: 93.664360 84.683617 67.141930
save ../result/submission/plot/car_detection_ground.txt
car_detection_ground AP: 41.456226 31.880611 24.876181
save ../result/submission/plot/car_detection_3d.txt
car_detection_3d AP: 29.933027 24.017235 18.650723
Your evaluation results are available at:
../result/submission
```

It only produce AP no AOS. Is it because you use testing set then evaluate directly on KITTI evaluation systems so they will provide you the detail result? Suppose we want to measure in Validation set, which value must I use? because there are many information such as `car_detection AP` `car_orientation AP` `car_detection_ground AP` `car_detection_3d AP`. So in specific how to produce table 1 as in paper if we use validation set?

-Thank you-

**Comment by Nicholasli1995** [comment]
> Hello @Nicholasli1995 thank you for your implementation. I have several questions.
> 
> 1. What split (train and validation split) did you use for this code? is it `subcnn` split?
> 2. during the inference using testing set do you use ROI from the object detection algorithm?
> 3. Is it still use the calibration or this system can automatically generate it?
> 4. How to produce the similar table as in paper? I see that you use AOS, AP, also something like Easy, Medium, Hard. However when I tried to use validation set evaluation and use `kitti_eval_offline` I only got some plots, and statistic informations in txt and final result as here.
> 
> ```
> python inference.py --cfg "../configs/try_KITTI_inference:test_submission.yml"
> 
> Wrote prediction file at ../result/gt_box_test/data/007025.txt
> Warning: 007025.png not included in detected images!
> PDFCROP 1.38, 2012/11/02 - Copyright (c) 2002-2012 by Heiko Oberdiek.
> ==> 1 page written on `car_detection.pdf'.
> PDFCROP 1.38, 2012/11/02 - Copyright (c) 2002-2012 by Heiko Oberdiek.
> ==> 1 page written on `car_orientation.pdf'.
> PDFCROP 1.38, 2012/11/02 - Copyright (c) 2002-2012 by Heiko Oberdiek.
> ==> 1 page written on `car_detection_ground.pdf'.
> PDFCROP 1.38, 2012/11/02 - Copyright (c) 2002-2012 by Heiko Oberdiek.
> ==> 1 page written on `car_detection_3d.pdf'.
> Thank you for participating in our evaluation!
> Loading detections...
> number of files for evaluation: 1634
>   done.
> save ../result/submission/plot/car_detection.txt
> car_detection AP: 93.934082 84.951851 67.594185
> save ../result/submission/plot/car_orientation.txt
> car_orientation AP: 93.664360 84.683617 67.141930
> save ../result/submission/plot/car_detection_ground.txt
> car_detection_ground AP: 41.456226 31.880611 24.876181
> save ../result/submission/plot/car_detection_3d.txt
> car_detection_3d AP: 29.933027 24.017235 18.650723
> Your evaluation results are available at:
> ../result/submission
> ```
> 
> It only produce AP no AOS. Is it because you use testing set then evaluate directly on KITTI evaluation systems so they will provide you the detail result? Suppose we want to measure in Validation set, which value must I use? because there are many information such as `car_detection AP` `car_orientation AP` `car_detection_ground AP` `car_detection_3d AP`. So in specific how to produce table 1 as in paper if we use validation set?
> 
> -Thank you-

1. The split files used in this study are detailed at https://github.com/Nicholasli1995/EgoNet/blob/master/docs/preparation.md and can be accessed at https://drive.google.com/drive/folders/1YLtptqspOFw08QG2MsxewDT9tjF2O45g.
2. Yes. This repo does not include object detection and any object detector can be used. My prepared detections for the test set are available at https://drive.google.com/drive/folders/1atfXLmsLFG6XEtNnwZuEYLydKqjr7Icf.
3. This systems uses KITTI camera intrinsic parameters. 
4. car_orientation AP is AOS. car_detection_ground AP is AP_BEV. car_detection AP is AP_2D.
